index.js

```diff
@@ -452,6 +455,21 @@ function trayCreate(json) {
     client.write(json.targetID, consts.eventNames.trayEventCreated)
 }
 
+// trayPopUpContextMenu pops up the context menu of the tray
+function trayPopUpContextMenu(json) {
+    let menu = menuCreate(json.menu);
+    if (!menu && !Object.keys(json.menuPopupOptions).length) {
```

Comment:
Logically, that should work according to the docs:
```typescript
popUpContextMenu(menu?: Menu, position?: Point)
```

However, even when I try in ElectronJS:
```javascript
 const handleClick = () => {
    tray.popUpContextMenu(undefined, undefined);
  };
```

I am getting the same error:
Uncaught Exception:
error processing argument at index 0, conversion failure from undefined at interface

That's why I put if else to check them properly.

Reply:
Out of curiosity, are you getting the same error when providing `null` instead of `undefined`?

Nevertheless that's a very weird behavior 🤔

Reply:
Yes, exactly the same error.

I already tried it before the commit, but here quick example to show:

```javascript
const handleClick = () => {
    tray.popUpContextMenu(null, null);
  };
```

Again:

Uncaught Exception:
TypeError: Error processing argument at index 0, conversion failure from null

PS: I am using Windows 10


I think it is related to electron.

Reply:
OK in that case I don't think this condition will work since the first argument of `.popUpContextMenu()` is the menu:

```javascript
if (!menu && typeof json.menuPopupOptions !== "undefined") {
    elements[json.targetID].popUpContextMenu(json.menuPopupOptions);
}
```

Therefore I would simplify `.trayPopUpContextMenu()` with:

```javascript
    let menu = menuCreate(json.menu);
    let position = json.menuPopupOptions;
    if (menu && position) elements[json.targetID].popUpContextMenu(menu, position);
    else if (menu) elements[json.targetID].popUpContextMenu(menu);
    else elements[json.targetID].popUpContextMenu();
```

What do you think? 🤔

Reply:
Actually, it will work but depend on your code:

1. If you created contextmenu, then it will show contextmenu with new position. --works
2. If you open popupcontextmenu manually without setting contextmenu, then it won't show anything. --won't work

Please see #Explanation for more details.

However, I agree that it will be much cleaner this way:

```javascript
// trayPopUpContextMenu pops up the context menu of the tray
function trayPopUpContextMenu(json) {
    let menu = menuCreate(json.menu);
    let position = json.menuPopupOptions;
    if (!menu && !position) elements[json.targetID].popUpContextMenu();
    else if (menu && !position) elements[json.targetID].popUpContextMenu(menu);
    else if (!menu && position) elements[json.targetID].popUpContextMenu(position);
    else elements[json.targetID].popUpContextMenu(menu, position);
}
```

I already committed this change.

## Explanation
 
- Here is the code with comments:

```javascript
function ToTray() {
  let tray = new Tray('icon.png');
  tray.setToolTip('This is my application.');
  const popUpContextMenu = Menu.buildFromTemplate([
    { label: 'item1', type: 'radio' },
    { label: 'item2', type: 'radio' },
    { label: 'item3', type: 'radio', checked: true },
    { label: 'item4', type: 'radio' }
  ]);
  //handleClick_Empty will open exactly contextMenu with default position (0,0)
  const handleClick_Empty = () => {
    tray.popUpContextMenu();
  };
  //handleClick_OnlyMenu can be a new menu (popUpContextMenu) instead of contextMenu with default position (0,0)
  const handleClick_OnlyMenu = () => {
    tray.popUpContextMenu(popUpContextMenu);
  };
  //handleClick_OnlyPos will display contextMenu with a new position
  const handleClick_OnlyPos = () => {
    tray.popUpContextMenu({ x: 500, y: 500 });
  };
  //handleClick_MenuPop will display a new menu (popUpContextMenu) with a new position
  const handleClick_MenuPop = () => {
    tray.popUpContextMenu(popUpContextMenu, { x: 500, y: 500 });
  };
  const contextMenu = Menu.buildFromTemplate([
    { label: 'empty', click: handleClick_Empty },
    { label: 'only menu', click: handleClick_OnlyMenu },
    { label: 'only pos', click: handleClick_OnlyPos },
    { label: 'menu-pop', click: handleClick_MenuPop }
  ]);

  tray.setContextMenu(contextMenu);

  // if I didn't setContextMenu and I wanted to open a menu manually (with right-click etc.)
  // tray.popUpContextMenu() --no error, but won't work -> since I didn't setup any contextmenu
  // tray.popUpContextMenu({ x: 500, y: 500 }) --no error, but won't work -> since I didn't setup any contextmenu
  // tray.popUpContextMenu(popUpContextMenu) --works
  // tray.popUpContextMenu(popUpContextMenu, {x:500, y:500}) --works
}
```

Please let me know if I am missing something.

Reply:
OK this is indeed a very weird behavior 🤔  I still don't get how `tray.popUpContextMenu({ x: 500, y: 500 });` would work 🤔 But your changes are fine 👍